scripts/duplicate_pr.py

```diff
@@ -0,0 +1,124 @@
+from enum import Enum
+
+import argparse
+import sh
+
+
+class BranchName(Enum):
+    MASTER = "master"
+    FORMPLAYER = "formplayer"
+
+
+def get_git(path=None):
+    return sh.git.bake(_tty_out=False, _cwd=path)
+
+
+def get_target_branch(orig_target_branch:str):
+    if orig_target_branch == BranchName.FORMPLAYER.value:
+        return BranchName.MASTER.value
+    elif orig_target_branch == BranchName.MASTER.value:
+        return BranchName.FORMPLAYER.value
+    else:
+        print("Only following branches are allowed: {}".format(", ".join([key.value for key in BranchName])))
+        exit(1)
+
+
+def git_create_branch(orig_branch_name:str, new_branch_name: str, git=None):
+    git = git or get_git()
+    try:
+        git.checkout(orig_branch_name)
+    except sh.ErrorReturnCode_1 as e:
+        print(red(e.stderr.decode()))
+        exit(1)
+    try:
+        git.checkout('-b', new_branch_name)
+    except sh.ErrorReturnCode_128 as e:
+        print(red(e.stderr.decode()))
+        exit(1)
+
+
+def git_pull_pr(pr_id:str, new_branch_name:str, git=None):
+    git = git or get_git()
+    input = "pull/{}/head:{}".format(pr_id,new_branch_name)
+    try:
+        git.fetch("origin", input)
+    except sh.ErrorReturnCode_1 as e:
+        print(red(e.stderr.decode()))
+        exit(1)
+
+
+def get_new_commits(base_branch: str, curr_branch:str, git=None):
+    git = git or get_git()
+    base_commit = merge_base_commit(base_branch, curr_branch)
+    recent_commit = latest_commit(curr_branch)
+
+    commits_range = "{}..{}".format(base_commit, recent_commit)
+    interested_commits = git("rev-list", "--no-merges", commits_range).split()
+    return interested_commits
+
+
+def cherry_pick_new_commits(commits:list[str], branch:str, git=None):
+    git = git or get_git()
+    git.checkout(branch)
+    for commits in reversed(commits):
+        git("cherry-pick", commits)
+
+
+def git_push_pr(branch:str, git=None):
+    git = git or get_git()
+    git.push("origin", branch)
+
+
+def merge_base_commit(branch1: str, branch2:str, git=None):
+    git = git or get_git()
+    return str(git("merge-base", branch1, branch2).replace("\n", ""))
+
+
+def latest_commit(branch:str, git=None):
+    git = git or get_git()
+    return str(git("rev-parse", branch).replace("\n", ""))
+
+
+def _wrap_with(code):
+
+    def inner(text, bold=False):
+        c = code
+
+        if bold:
+            c = "1;%s" % c
+        return "\033[%sm%s\033[0m" % (c, text)
+    return inner
+
+
+red = _wrap_with('31')
+
+
+def main():
+    parser = argparse.ArgumentParser(description='Duplicate and push PR between formplayer/master')
+    parser.add_argument('orig_pr_id', type=str, help="ID of PR to be cloned")
```

Comment:
do we need to have PR ID if we have original branch name ? Can't we just pull the commits using branch name ?

Reply:
I'm using the PR ID to pull the latest PR branch from remote in cases where that branch doesn't exist locally. Are you saying there's a way to do this with just the branch name?

Reply:
yeah, we can pull the branch directly from remote - 

````
git fetch origin/branch_name
git checkout branch_name
````

Reply:
Thanks for the tip - [5e34e42](https://github.com/dimagi/commcare-core/pull/1254/commits/5e34e42c75e16722f05e19cc8b544e490e37a748)

And I think it's `git fetch origin branch_name`